coa/pkg/apis/v1alpha2/providers/pubsub/redis/redis.go

```diff
@@ -198,40 +213,48 @@ func (i *RedisPubSubProvider) worker() {
 	}
 }
 func (i *RedisPubSubProvider) processMessage(msg RedisMessageWrapper) error {
+	i.ClaimedMessages[msg.MessageID] = true
 	var evt v1alpha2.Event
 	err := json.Unmarshal([]byte(utils.FormatAsString(msg.Message)), &evt)
 	if err != nil {
 		return v1alpha2.NewCOAError(err, "failed to unmarshal event", v1alpha2.InternalError)
 	}
 	if err := msg.Handler(msg.Topic, evt); err != nil {
+		delete(i.ClaimedMessages, msg.MessageID)
```

Comment:
This operation should be protected by TopicLock as well. Also, I think we may want to distinguish non-recoverable errors and recoverable errors like transient errors, and auto-delete only failures caused by non-recoverable and failures that we know that will be retried by upper layers.

Reply:
You are right. in memory map is not thread safe. I have put the operation in the lock. 
For subscriber error handling, we have another issue tracking it. https://github.com/eclipse-symphony/symphony/issues/306. In this PR, I only change the job subscriber to always return no error since we have reconciler for job deployment.